### PR TITLE
Valkyrie publish file.characterized event for derivative creation

### LINKED ISSUE #6162: ValkyrieCharacterizationService should publish file.characterized
### Descriptive summary

Currently, the ValkyrieIngestJob enqueues the derivatives job directly, and enqueues the characterization job via an `object.file.uploaded` event.

Instead, ValkyrieIngestJob should not enqueue ValkyrieCreateDerivativesJob. ValkyrieCharacterizationService should publish file.characterized and FileMetadataListener should enqueue either CreateDerivativesJob or ValkyrieCreateDerivativesJob depending on the payload's type.

See https://github.com/samvera/hyrax/blob/9d06f2b4cc23138259b2703f575a8494a7aa43d5/app/services/hyrax/listeners/file_metadata_listener.rb#L15

### Acceptance Criteria/Expected Behavior

This is a code refactor, but may fix race condition bugs if any exist when derivatives try to run before characterization is finished.

### PR DESCRIPTION
**Pull request by lsitu**
### Fixes

Fixes #6162 ; refs #6162

### Summary

Valkyrie publish file.characterized event for derivative creation.

@samvera/hyrax-code-reviewers

### CODE REVIEW

File: app/services/hyrax/listeners/file_metadata_listener.rb [hunk 1]
```diff
@@ -12,8 +12,20 @@ class FileMetadataListener
...
-        CreateDerivativesJob
-          .perform_later(event[:file_set], event[:file_id], event[:path_hint])
+        file_set = event[:file_set]
+
+        case file_set
+        when ActiveFedora::Base # ActiveFedora
+          CreateDerivativesJob
+            .perform_later(file_set, event[:file_id], event[:path_hint])
+        else
+          # With valkyrie, file.characterized event is also published for derivatives.
```
Comment (no-reply) [inline app/services/hyrax/listeners/file_metadata_listener.rb:22]: i think we don't want this to be true!

the valkyrie code path shouldn't be calling characterization on files that don't require characterization, right? i think this filter should go below around https://github.com/samvera/hyrax/pull/6180/files#diff-c1500de3cd457700a4445ea1b341164b5a822aa64a086d85e12eb9adfdfeba79R54, or else in the `ValkyrieCharacterizationService`?

Comment (no-reply) [reply app/services/hyrax/listeners/file_metadata_listener.rb]: >  But in UCSD dams, currently we have file characterization for all files, including derivatives.

that seems okay. if downstream applications want to characterize under different conditions, it's easy to subscribe a new listener with the desired behavior.

File: app/services/hyrax/listeners/file_metadata_listener.rb [hunk 2]
```diff
@@ -39,7 +47,9 @@ def on_file_metadata_updated(event)
       # @param [Dry::Events::Event] event
       # @return [void]
       def on_file_uploaded(event)
-        # Run characterization
+        # Run characterization for original file only
+        return unless event[:metadata].original_file?
```
Comment (no-reply) [inline app/services/hyrax/listeners/file_metadata_listener.rb:51]:
```suggestion
        return unless event[:metadata]&.original_file?
```

`:metadata` probably SHOULD always be provided, but unhandled exceptions in listeners are very disruptive, so it's nice not to allow for `nil` errors.

**Review by no-reply — APPROVED** [review]
looks good, minus the minor suggestion, which i won't block on.

**Review by dunn — APPROVED** [review]
tested successfully locally

**Pull request merged**